app/controllers/reservations_controller.rb

```diff
@@ -282,9 +283,14 @@ def load_basic_resources
     @order = Order.find(params[:order_id])
     # It's important that the order_detail be the same object as the one in @order.order_details.first
     @order_detail = @order.order_details.find { |od| od.id.to_i == params[:order_detail_id].to_i }
```

Comment:
This seems weird. Do you know why we're doing this weird find rather than a `find(params[:order_detail_id]` or a `find_by(id: params[:order_detail_id])`?